### PR TITLE
complain feature added

### PR DESCRIPTION
**Pull request by Deven-Rathore**
test complain feature

### CODE REVIEW

File: src/main/java/com/pytosoft/controller/ComplainController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	private ComplainService complainService;
+
+	@GetMapping("/getAll")
+	public List<Complain> getAllComplain() {
+		return complainService.listAll();
+	}
+
+	@PostMapping("/addComplain")
+	public String add(@RequestBody Complain complain) {
+		complainService.save(complain);
```
Comment (csrathore617) [inline src/main/java/com/pytosoft/controller/ComplainController.java:33]: Change return type to responceentity

File: src/main/java/com/pytosoft/controller/ComplainController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	@Autowired
+	private ComplainService complainService;
+
+	@GetMapping("/getAll")
+	public List<Complain> getAllComplain() {
+		return complainService.listAll();
+	}
+
+	@PostMapping("/addComplain")
+	public String add(@RequestBody Complain complain) {
```
Comment (shekhawatvikram) [inline src/main/java/com/pytosoft/controller/ComplainController.java:32]: change return type to ResponseEntity

Comment (Deven-Rathore) [reply src/main/java/com/pytosoft/controller/ComplainController.java:32]: ok got it

**Review by csrathore617 — COMMENTED** [review]
change return type to responceentity in complain controller

**Review by shekhawatvikram — COMMENTED** [review]
change it

File: src/main/java/com/pytosoft/controller/ComplainController.java [hunk 3]
```diff
@@ -46,13 +46,13 @@ public ResponseEntity<Complain> getComplainById(@PathVariable Integer id) {
...
-	@DeleteMapping("/{id}")
+	@DeleteMapping("/comp/{id}")
 	public String delete(@PathVariable Integer id) {
 		complainService.deleteComplain(id);
 		return " complain sucessfully deleted" + id;
 	}
 
-	@DeleteMapping("/{id}")
+	@DeleteMapping("/obcomp/{id}")
 	public String delete(@PathVariable Long id) {
```
Comment (csrathore617) [inline src/main/java/com/pytosoft/controller/ComplainController.java:56]: change return type to responceentity

File: src/main/java/com/pytosoft/controller/MedicineController.java [hunk 4]
```diff
@@ -46,10 +46,10 @@ public ResponseEntity<Medicine> getMedicineById(@PathVariable Long id) {
 		}
 	}
 
-	@DeleteMapping("/{id}")
-	public String delete(@PathVariable Long id) {
+	@DeleteMapping("/observed/{id}")
```
Comment (csrathore617) [inline src/main/java/com/pytosoft/controller/MedicineController.java:49]: responceentity?

File: src/main/java/com/pytosoft/controller/ComplainController.java [hunk 5]
```diff
@@ -46,13 +46,13 @@ public ResponseEntity<Complain> getComplainById(@PathVariable Integer id) {
...
 		complainService.deleteObservedComplain(id);
```
Comment (csrathore617) [inline src/main/java/com/pytosoft/controller/ComplainController.java:57]: change return type to responceentity

**Review by csrathore617 — APPROVED** [review]
Tested

**Review by csrathore617 — APPROVED** [review]

**Review by csrathore617 — APPROVED** [review]
Tested

File: src/main/java/com/pytosoft/controller/ComplainController.java (hunk 1 above)
Comment (Deven-Rathore) [reply src/main/java/com/pytosoft/controller/ComplainController.java:33]: ok got it

**Review by shekhawatvikram — APPROVED** [review]

File: src/main/java/com/pytosoft/service/ComplainService.java [hunk 6]
```diff
@@ -17,9 +19,14 @@ public class ComplainService {
     @Autowired
     private ObservedComplainRepository observedComplainRepository;
     
+    @Autowired
+    private DoctorRepo repo;
```
[the remark this answers is not in the document]
Comment (Deven-Rathore) [reply src/main/java/com/pytosoft/service/ComplainService.java]: ok got it

**Pull request merged**